src/lib/resources/auth.resource.ts

```diff
@@ -1,6 +1,14 @@
+import { ApiService } from '../api/api.service';
+import { ArchiveResponse } from '../api/archive.repo';
+import { PermSdkError } from '../error';
+
+import { ArchiveStore } from './archive';
 import { BaseResource } from './base.resource';
 
 export class AuthResource extends BaseResource {
+  constructor(public api: ApiService, public archiveStore: ArchiveStore) {
```

Comment:
I'm trying to fully understand why the authentication resource relies on archive endpoints -- is it possible to fully decouple the two? I understand that we need a "default" (or "base") archive for other calls, but that doesn't feel like the responsibility of the authentication object.  Would it make sense for this to be in a higher level coordinating object would need to handle and pass to specific resources as needed.

Reply:
It's a little less clear without the changes that come after this that show everything working together a bit more, will try to explain.

So there's a few different types of entities at play here: 
- `Store`
- `Repo`
- `Resource`

A `Store` is just some shared state that just holds data that can be accessed via some getters and setters, to be injected into other things as needed. Perhaps the `ArchiveStore` should be called the `SessionStore`, it's meant to just hold some data pertaining to the session: the current `archive` and the root folder of that `archive`.

A `Repo` is meant to just be a wrapper for API calls, formatting inputs properly and returning the response. No state, no nothing, just user input -> API output while handling the boilerplate of making those requests.

A `Resource` allows for a bit more internal plumbing to expose a more simplified interface for the end user to interact with to accomplish a specific task, potentially chaining multiple API calls to accomplish the task while dealing with the API responses to return only the specific data relevant to the task, abstracting away any interaction with the full API response. 

So, as far as 
> the authentication resource relies on archive endpoints 

I wouldn't say that it does, it just requires access to the shared state in the `ArchiveStore` to update the current archive while calling `useArchive` after getting a successful API response.